File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int i = 0,j = 0,sum = 0;
        double max_sum = Double.NEGATIVE_INFINITY;
        while (j < nums.length){
            sum += nums[j];
            if ((j - i + 1) < k){
                j ++;
            }
            else if ((j - i + 1) == k){
                max_sum = Math.max(max_sum,sum);
                sum -= nums[i];
                i++;j++;
            }
        }
        return max_sum / k;
    }
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution(object):
    def findMaxAverage(self, nums, k):
        left,right,max_avg,sum_ele = 0,0,float('-inf'),0
        while right < len(nums):
            sum_ele += nums[right]
            if (right - left + 1) < k:
                right += 1
            elif (right - left + 1) == k:
                max_avg = max(max_avg,sum_ele)
                sum_ele -= nums[left]
                left += 1
                right += 1
        return float(max_avg) / float(k)

        